src/client/components/LoginPage/index.jsx

```diff
@@ -237,8 +237,23 @@ const LoginPage = ({
     )
   }
 
-  // User is logged in, redirect if available
   if (location) {
+    // Nested if conditions to prevent undefine error from object and undefine checks
+    // un-parsed object will give you true even when the proproperty is undefined
+    const parsedLocation = JSON.parse(JSON.stringify(location))
+    if ('state' in parsedLocation) {
+      const { state } = location
+      if (state) {
+        const parsedState = JSON.parse(JSON.stringify(state))
+        if ('previous' in parsedState) {
+          const { previous } = state
+          if (previous === '/directory') {
```

Comment:
Flagging this as code smell - could one use `location?.state?.previous?` instead?